Gemfile

```diff
@@ -26,6 +26,10 @@ platforms :ruby do
   gem "sqlite3",                "~> #{sqlite3_version}"
   # seamless_database_pool requires Ruby ~> 2.0
   gem "seamless_database_pool", "~> 1.0.20" if Gem::Version.new(RUBY_VERSION) < Gem::Version.new('3.0.0')
+  gem "trilogy" if version >= 6.0
+  if version >= 6.0 && version <= 7.0
+    gem "activerecord-trilogy-adapter"
+  end
```

Comment:
Regarding the integration of Trilogy, theoretically, it should seamlessly function with any framework equipped with the `active record-trilogy-adapter`. Notably, Rails 7.1 conveniently incorporates this adapter by default. You can explore the prerequisites for this gem [here](https://github.com/trilogy-libraries/activerecord-trilogy-adapter?tab=readme-ov-file#requirements). However, a noteworthy discovery I made is the importance of the `composite_primary_keys` version. Compatibility issues arise with versions preceding AR 7 due to limitations intertwined with the ActiveRecord version.